File: algorithms/Base7/solution.py
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        res = ''
        is_negative = False
        
        if num == 0:
            res = '0'
        if num < 0:
            num = -num
            is_negative = True
        while num > 0:
            res = str(num % 7) + res
            num = num / 7
        return '-' + res if is_negative else res

File: algorithms/Base7/Solution.java
    class Solution {
    public String convertToBase7(int num) {
        String res = "";
        boolean isNegative = false;
        
        if (num == 0) {
            res = "0";
        } else {
            if (num < 0) {
                num = -num;
                isNegative = true;
            }
            
            while (num > 0) {
                res = String.valueOf(num % 7) + res;
                num /= 7;
            }
        }
        
        return isNegative ? "-" + res : res;
    }
}